llm_bioinformatics_research/src/UserProfile/UserProfile.js

```diff
@@ -62,7 +62,17 @@ const UserProfile = () => {
     if (file) {
       const reader = new FileReader();
       reader.onloadend = () => {
-        setUser({ ...user, image: reader.result });
+        const img = new Image();
+        img.src = reader.result;
+        img.onload = () => {
+          const canvas = document.createElement('canvas');
+          const ctx = canvas.getContext('2d');
+          canvas.width = 100;
+          canvas.height = 100;
+          ctx.drawImage(img, 0, 0, 100, 100);
+          const resizedImage = canvas.toDataURL('image/png');
+          setUser({ ...user, image: resizedImage });
```

Comment:
The image is also being saved with the correct name so well done. 
<img width="806" alt="Screenshot 2024-10-28 at 8 31 48 PM" src="https://github.com/user-attachments/assets/42867c7d-8e94-4836-b242-a673a9846e3c">